codalab/lib/bundle_cli.py

```diff
@@ -1746,8 +1746,11 @@ def do_detach_command(self, args):
                             detach = True
             if not detach:
                 new_items.append(item)
-
-        client.create('worksheet-items', data=new_items, params={'replace': True})
+        if len(new_items) > 0:
```

Comment:
`cl detach` last item failed when the `new_items` array is empty. So here I added some logic to handle the rest call when it sends an empty array over.

Reply:
I don't understand what `new_items` has to do with passing in `uuid`.

Reply:
When I was debugging the problem worksheet, I noticed that the `new_items` is always `empty` when it passed to `client.fetch`. Then in `client.fetch` side, there is a for loop on `new_items` to generate `worksheet_to_items`. Since `new_items` is empty, `worksheet_to_items` is also empty, then the logic of checking the `replace` variable will never be executed.
So I tried to split the `replace` logic from original for loop. Then I feel if the request is looking for replacing a worksheet with no data inside, we just need to update the worksheet with empty data. Hence, I pass an empty list to `update_worksheet_items` function.